charmcraft/application/commands/store.py

```diff
@@ -1936,87 +1937,95 @@ def run(self, parsed_args):
             architectures = ["all"]
 
         if parsed_args.filepath:
-            resource_filepath = parsed_args.filepath
-            resource_filepath_is_temp = False
-            resource_type = ResourceType.file
-            emit.progress(f"Uploading resource directly from file {str(resource_filepath)!r}.")
+            emit.progress(f"Uploading resource directly from file {str(parsed_args.filepath)!r}.")
             bases = [{"name": "all", "channel": "all", "architectures": architectures}]
+            result = store.upload_resource(
+                parsed_args.charm_name,
+                parsed_args.resource_name,
+                ResourceType.file,
+                parsed_args.filepath,
+                bases=bases,
+            )
         elif parsed_args.image:
+            emit.progress("Getting image")
+            emit.debug("Trying to get image from Docker")
+            image_service = self._services.image
+            if digest := image_service.get_maybe_id_from_docker(parsed_args.image):
+                emit.debug("Image is available via the local Docker daemon.")
+                source_path = f"docker-daemon:{digest}"
+            elif (image_path := pathlib.Path(parsed_args.image)).exists():
```

Comment:
should you do this check first? It seems cheaper than checking docker

Reply:
It is cheaper, but I don't want to break backwards compatibility here.

Reply:
makes sense! Should we add a note about backwards-compat here?

Reply:
Done!